h2o-core/src/test/java/water/rapids/ast/prims/assign/AstRectangleAssignTest.java

```diff
@@ -64,6 +82,47 @@ public void testAssignFrameSlice_domainsDiffer() throws Exception {
     }
   }
 
+  @Test
+  public void testAssignFrameSlice_strings() throws Exception {
+    Vec strVec = seqStrVec(10000, 5, 7, 8, 10000);
+    final Frame data = new Frame(Key.<Frame>make("data"), null, new Vec[] {strVec});
+    DKV.put(data._key, data);
+    Frame output = null;
+    try {
+      String rapids = "(tmp= tst (:= data (rows (cols data [0.0] ) [10000.0:" + _nRows + ".0] ) [0.0] [0.0:" + _nRows + ".0] ) )";
+      Val val = Rapids.exec(rapids);
+      if (val instanceof ValFrame) {
+        output = val.getFrame();
+        String[] expected = strVec2array(strVec);
+        System.arraycopy(expected, 10000, expected, 0, _nRows);
+        String[] actual = strVec2array(output.vec(0));
+        assertArrayEquals(expected, actual);
+      }
+    } finally {
+      strVec.remove();
+      data.delete();
+      if (output != null) {
+        output.delete();
+      }
+    }
+  }
+
+  private static Vec seqStrVec(int... runs) {
+    Key k = Vec.VectorGroup.VG_LEN1.addVec();
```

Comment:
oki, I checked the code in Vec - `/** The common shared vector group for very short vectors */` - it will work, but the group could contain the vectors of different length => then it lose its purpose (VG represents compatible vectors). But that's different issue.